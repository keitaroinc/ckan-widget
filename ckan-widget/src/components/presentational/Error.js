import React, { Component } from 'react'

class Error extends Component {
    constructor() {
        super()
        this.state = { show: true }
    }

    handleClose = e => {
        this.setState({ show: false })
    }

	render() {
        if (this.state.show) {
            return(
                <div className="my-3 alert alert-warning alert-dismissible" role="alert">
                    <strong>Upps, error!</strong> {this.props.error};
                    <button type="button" className="close" data-dismiss="alert" aria-label="Close" onClick={this.handleClose}>
                        <span aria-hidden="true">&times;</span>
                    </button>
                </div>
            )
        }

        return(null)
  	}
}

export default Error
