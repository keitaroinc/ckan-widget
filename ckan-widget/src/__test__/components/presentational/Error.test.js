import React from 'react';
import { shallow } from 'enzyme';
import Error from '../../../components/presentational/Error';

let component

describe('Error', () => {
  beforeEach(() => {
    component = shallow(
      <Error error={'Network error!'} />
    );
  })

  it('should be defined', () => {
    expect(Error).toBeDefined();
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should be shown at the begging', () => {
    expect(component.find('div.alert').length).toEqual(1);
  })

  it('should close the alert', () => {
    expect(component.instance().state.show).toBe(true)

    const alert = component.find('button')
    alert.simulate('click')
    expect(component.instance().state.show).toBe(false)
    expect(component.find('div.alert').length).toEqual(0);
  })
});
