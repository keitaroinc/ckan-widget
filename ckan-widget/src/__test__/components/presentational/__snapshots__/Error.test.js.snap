// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    error="Network error!"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <strong>
          Upps, error!
</strong>,
        " ",
        "Network error!",
        <button
          aria-label="Close"
          className="close"
          data-dismiss="alert"
          onClick={[Function]}
          type="button"
>
          <span
                    aria-hidden="true"
          >
                    ×
          </span>
</button>,
      ],
      "className": "my-3 alert alert-warning alert-dismissible",
      "role": "alert",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Upps, error!",
        },
        "ref": null,
        "rendered": "Upps, error!",
        "type": "strong",
      },
      " ",
      "Network error!",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Close",
          "children": <span
            aria-hidden="true"
>
            ×
</span>,
          "className": "close",
          "data-dismiss": "alert",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "children": "×",
          },
          "ref": null,
          "rendered": "×",
          "type": "span",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <strong>
            Upps, error!
</strong>,
          " ",
          "Network error!",
          <button
            aria-label="Close"
            className="close"
            data-dismiss="alert"
            onClick={[Function]}
            type="button"
>
            <span
                        aria-hidden="true"
            >
                        ×
            </span>
</button>,
        ],
        "className": "my-3 alert alert-warning alert-dismissible",
        "role": "alert",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Upps, error!",
          },
          "ref": null,
          "rendered": "Upps, error!",
          "type": "strong",
        },
        " ",
        "Network error!",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Close",
            "children": <span
              aria-hidden="true"
>
              ×
</span>,
            "className": "close",
            "data-dismiss": "alert",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": "×",
            },
            "ref": null,
            "rendered": "×",
            "type": "span",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
